src/main/java/net/wovenmc/woven/mixin/item/settings/MixinAbstractFurnaceBlockEntity.java

```diff
@@ -26,26 +28,60 @@
 
 import net.minecraft.block.entity.AbstractFurnaceBlockEntity;
 import net.minecraft.item.Item;
+import net.minecraft.item.ItemConvertible;
 import net.minecraft.item.ItemStack;
+import net.minecraft.item.Items;
+import net.minecraft.recipe.Recipe;
+import net.minecraft.util.Identifier;
+import net.minecraft.util.collection.DefaultedList;
+import net.minecraft.util.registry.Registry;
 
 @Mixin(AbstractFurnaceBlockEntity.class)
 public abstract class MixinAbstractFurnaceBlockEntity {
+	@Shadow
+	protected DefaultedList<ItemStack> inventory;
+	private static final Identifier FUEL_ID = new Identifier("furnace_fuel");
 	private final ThreadLocal<ItemStack> stack = new ThreadLocal<>();
 
 	@Inject(method = "tick", at = @At(value = "INVOKE", target = "Lnet/minecraft/item/Item;getRecipeRemainder()Lnet/minecraft/item/Item;"),
 			locals = LocalCapture.CAPTURE_FAILEXCEPTION)
-	private void grabLocalStack(CallbackInfo info, boolean isBurning, boolean isCooking, ItemStack cookingStack) {
+	private void grabLocalStack(CallbackInfo info, boolean isBurning, boolean isCooking, ItemStack cookingStack, Recipe<?> recipe) {
 		stack.set(cookingStack);
 	}
 
 	@Redirect(method = "tick", at = @At(value = "INVOKE", target = "Lnet/minecraft/item/Item;getRecipeRemainder()Lnet/minecraft/item/Item;"))
-	private Item getNewRemainder(Item origItem, Item origReturn) {
+	private Item hackCustomFuelRemainder(Item origItem) {
 		WovenItemSettingsHolder holder = (WovenItemSettingsHolder) origItem;
+		System.out.println(Registry.ITEM.getId(origItem));
```

Comment:
Might not want that for when it will be merged. :)

Reply:
oops